CMR/python/design.md

```diff
@@ -0,0 +1,42 @@
+# CMR Python Wrapper Design
+
+Design concepts for the code.
+
+## Overview
+
+The CMR Python Wrapper was created to be a wrapper around the CMR API and to be a bridge between Jypter Notebooks and CMR.
+
+## Code Conventions
+
+### Lambdas
+When ever there is a choice on how to process some data or which "method" should be used for some algorithm, then these choices should be captured as lambda functions. Lambda functions can defer processing to the point where it is actually needed (or skipped if it no longer makes since to run).
+
+    def search(query = None,
+    	filters = None,
+		limit = None,
+		config: dict = None):
+		
+For the search function in cmr/search/collection.py, the second parameter is actually a list. For filter it is a list of lambda functions which reduced the number of fields returned. These filters can then be applied in any order and can be swapped out as needed by the user in a notebook. These filters should be short and very specific, handling only one item at a time. In this way the calling code can figure out how best to loop through the set of items, especially in the case where function generators are to be used and data is not processed till it is needed, at which time lambda functions can be executed on demand using the smallest chuck of data available.
```

Comment:
Long line, so going to suggest edits for this sentence from "For filter it is a list of lambda functions which reduced the number of fields returned."
to: "For `filters` it is a list of lambda functions which reduce the number of fields returned."